lib/game_client/ogs/ogs_game_client.dart

```diff
@@ -153,15 +252,114 @@ class OGSGameClient extends GameClient {
+    final speed = parts.length >= 2 ? parts[1] : 'rapid';
+
+    // Generate a UUID for the automatch request and store it for later cancellation
+    _currentAutomatchUuid = const Uuid().v4();
+    _automatchCompleter = Completer<Game>();
+
+    // Set up message listener for automatch responses
+    _messageSubscription =
+        _webSocketManager.messages.listen(_handleAutomatchResponse);
+
+    // Create the automatch payload data matching OGS format
+    final automatchData = {
+      "uuid": _currentAutomatchUuid!,
+      "size_speed_options": [
+        {"size": "${boardSize}x$boardSize", "speed": speed, "system": "byoyomi"}
+      ],
+      "lower_rank_diff": 3,
+      "upper_rank_diff": 3,
+      "rules": {"condition": "required", "value": "japanese"},
+      "handicap": {"condition": "preferred", "value": "enabled"}
+    };
+
+    // Send the automatch request via WebSocket
+    _webSocketManager.send('automatch/find_match', automatchData);
+
+    return _automatchCompleter!.future;
+  }
+
+  void _handleAutomatchResponse(Map<String, dynamic> message) async {
+    // Check if this is an automatch/start message
+    if (message['event'] == 'automatch/start' &&
+        message['data']['uuid'] == _currentAutomatchUuid) {
+      final gameId = message['data']['game_id'] as int;
+
+      // Clean up resources
+      _messageSubscription?.cancel();
+      _messageSubscription = null;
+      _currentAutomatchUuid = null;
+
+      final response = await _httpClient.get(
+        Uri.parse('$serverUrl/api/v1/games/$gameId'),
+        headers: {
+          'User-Agent': _userAgent,
+          if (_csrfToken != null) 'X-CSRFToken': _csrfToken!,
+        },
+      );
+
+      Map<String, dynamic> gameData = {};
+      if (response.statusCode == 200) {
+        gameData = jsonDecode(response.body)['gamedata'];
+      } else {
+        throw Exception('Failed to fetch game details: ${response.statusCode}');
+      }
+
+      try {
+        final game = OGSGame(
+          id: gameId.toString(),
+          boardSize: gameData['width'] as int? ?? 19,
+          rules: Rules.japanese,
+          handicap: gameData['handicap'] as int? ?? 0,
+          komi: gameData['komi'] as double? ?? 6.5,
+          myColor: gameData['players']['black']['id'].toString() ==
+                  _userInfo.value?.userId
+              ? wq.Color.black
+              : wq.Color.white,
+          timeControl: TimeControl(
+            mainTime: Duration(
+                seconds: gameData['time_control']?['main_time'] as int? ?? 300),
+            timePerPeriod: Duration(
+                seconds:
+                    gameData['time_control']?['period_time'] as int? ?? 30),
+            periodCount: gameData['time_control']?['periods'] as int? ?? 5,
+          ),
+          previousMoves: [],
+          webSocketManager: _webSocketManager,
+        );
+
+        // TODO: there's chance of a race where the automatch may have been created, but
```

Comment:
have you run into this in practice? I guess the safest is to ignore automatchs for locally cancelled requests (if that doesn't forfeit the game).